tasks/kernel_matrix_testing/compiler.py

```diff
@@ -118,7 +118,7 @@ def exec(self, cmd: str, user="compiler", verbose=True, run_dir: PathOrStr | Non
         self.ensure_running()
 
         # Set FORCE_COLOR=1 so that termcolor works in the container
-        self.ctx.run(
+        return self.ctx.run(
```

Comment:
What does this accomplish?

Reply:
In [`ensure_ready_for_cross_compile`](https://github.com/DataDog/datadog-agent/blob/main/tasks/kernel_matrix_testing/compiler.py#L196) we check the result of a command ran on the container with `exec()`, but the result was always None as this function was not returning anything, so it was always preparing the cross-compile environment. Adding the return fixes that and now only runs the setup if needed.